src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -983,12 +983,10 @@ private void ProcessDependencies(Packaging.PackageBuilder builder)
             }
             else
             {
-                // TO FIX: when we persist the target framework into packages.config file,
-                // we need to pull that info into building the PackageDependencySet object
                 builder.DependencyGroups.Clear();
```

Comment:
It appears this comment was written before `packages.config` files had the `targetFramework` attribute on each package. The way the implementation ended up, I disagree with the comment.

When you retarget a project to a different version of .NET Framework, nothing automatically updates packages.config to set all the `targetFramework` attributes to the current project version. Therefore, it's not uncommon to have pacakges.config files that multiple unique versions of .NET across the different packages. The project does not multi-target, and the package only contains a single `lib/<tfm>/project.dll`, so it's objectively incorrect for the package being created to have multiple dependency groups for different TFMs. Even if all the TFMs in the packages.config file are the same, if it's different to the lib asset, then it might not be correct either.